src/sentry/users/api/serializers/user.py

```diff
@@ -0,0 +1,388 @@
+from __future__ import annotations
+
+import itertools
+import warnings
+from collections import defaultdict
+from collections.abc import Mapping, MutableMapping, Sequence
+from datetime import datetime
+from typing import Any, DefaultDict, TypedDict, cast
+
+from django.conf import settings
+from django.contrib.auth.models import AnonymousUser
+
+from sentry import experiments
+from sentry.api.serializers import Serializer, register
+from sentry.api.serializers.types import SerializedAvatarFields
+from sentry.app import env
+from sentry.auth.elevated_mode import has_elevated_mode
+from sentry.hybridcloud.services.organization_mapping import organization_mapping_service
+from sentry.models.authidentity import AuthIdentity
+from sentry.models.organization import OrganizationStatus
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.models.organizationmembermapping import OrganizationMemberMapping
+from sentry.organizations.services.organization import RpcOrganizationSummary
+from sentry.users.models.authenticator import Authenticator
+from sentry.users.models.user import User
+from sentry.users.models.user_avatar import UserAvatar
+from sentry.users.models.user_option import UserOption
+from sentry.users.models.useremail import UserEmail
+from sentry.users.models.userpermission import UserPermission
+from sentry.users.models.userrole import UserRoleUser
+from sentry.users.services.user import RpcUser
+from sentry.utils.avatar import get_gravatar_url
+from sentry.utils.serializers import manytoone_to_dict
+
+
+class _UserEmails(TypedDict):
+    id: str
+    email: str
+    is_verified: bool
+
+
+class _Organization(TypedDict):
+    slug: str
+    name: str
+
+
+class _Provider(TypedDict):
+    id: str
+    name: str
+
+
+class _Identity(TypedDict):
+    id: str
+    name: str
+    organization: _Organization
+    provider: _Provider
+    dateVerified: datetime
+    dateSynced: datetime
+
+
+class _UserOptions(TypedDict):
+    theme: str  # TODO: enum/literal for theme options
+    language: str
+    stacktraceOrder: int  # TODO enum/literal
+    defaultIssueEvent: str
+    timezone: str
+    clock24Hours: bool
+    prefersIssueDetailsStreamlinedUI: bool
+
+
+class UserSerializerResponseOptional(TypedDict, total=False):
+    # NOTE: There is a bug here where trying to move these fields to
+    # UserSerializerResponse and using NotRequired. "identities" is marked as
+    # required for places where UserSerializerResponse is used as a field (e.g
+    # OrganizationMemberResponse).
+    identities: list[_Identity]
+    avatar: SerializedAvatarFields
+    authenticators: list[Any]  # TODO: find out what type this is
```

Comment:
Interesting that we have a special serialization format for authenticators here that is different from the representation that authenticator endpoints use. 🤔

Reply:
yeahh..... the `user_authenticator_enroll.py` endpoint also has it's own serializer /format for each of the interfaces that's different. AuthenticatorInterface serialization in general is like p. inconsistent ;-;